x-pack/plugins/transform/public/app/sections/transform_management/components/edit_transform_flyout/use_edit_transform_flyout.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEqual } from 'lodash';
+import { useReducer } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { TransformPivotConfig } from '../../../../common';
+
+const stringNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormStringNotValidErrorMessage',
+  {
+    defaultMessage: 'Value needs to be of type string.',
+  }
+);
+
+// The way the current form is set up,
+// this validator is just a sanity check,
+// it should never trigger an error.
+const stringValidator = (arg: any): string[] =>
+  typeof arg === 'string' ? [] : [stringNotValidErrorMessage];
+
+const frequencyNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormFrequencyNotValidErrorMessage',
+  {
+    defaultMessage: 'The frequency value is not valid.',
+  }
+);
+
+// Only allow frequencies in the form of 1s/1h etc.
+// Note this doesn't do a check against 0s yet.
+const frequencyValidator = (arg: any): string[] => {
+  return /^([0-9]+[d|h|m|s|])$/.test(arg) ? [] : [frequencyNotValidErrorMessage];
+};
+
+const validate = {
+  string: stringValidator,
+  frequency: frequencyValidator,
+};
+
+interface Field {
+  errorMessages: string[];
+  isOptional: boolean;
+  validator: keyof typeof validate;
+  value: string;
+}
+
+const defaultField: Field = {
+  errorMessages: [],
+  isOptional: true,
+  validator: 'string',
+  value: '',
+};
+
+interface EditTransformFlyoutFieldsState {
+  [key: string]: Field;
+  description: Field;
+  frequency: Field;
+}
+
+export interface EditTransformFlyoutState {
+  formFields: EditTransformFlyoutFieldsState;
+  isFormTouched: boolean;
+  isFormValid: boolean;
+}
+
+// This is not a redux type action,
+// since for now we only have one action type.
+interface Action {
+  field: keyof EditTransformFlyoutFieldsState;
+  value: string;
+}
+
+// Some attributes can have a value of `null` to trigger
+// a reset to the default value.
+interface UpdateTransformPivotConfig {
+  description: string;
+  frequency: string;
+}
+
+// Takes in the form configuration and returns a
+// request object suitable to be sent to the
+// transform update API endpoint.
+export const applyFormFieldsToTransformConfig = (
+  config: TransformPivotConfig,
+  { description, frequency }: EditTransformFlyoutFieldsState
+): Partial<UpdateTransformPivotConfig> => {
+  const updateConfig: Partial<UpdateTransformPivotConfig> = {};
+
+  if (!(config.frequency === undefined && frequency.value === '')) {
```

Comment:
nit, for readability, i think this would be clearer as 
`if (config.frequency !== undefined && frequency.value !== '') {`

Reply:
I updated the validation and added jest tests for it in dedd678. Note that negating both checks gives different results than just doing each individual check (esp. now with one more additional check), the unit tests now cover this. It's necessary because the form always will return a string based value (e.g. `''`) but the actual config for example for `description` could be either not set at all or explicitly set to an empty string.